stdlib/zipfile.pyi

```diff
@@ -24,13 +35,53 @@ class ZipExtFile(io.BufferedIOBase):
     mode: str
     name: str
     if sys.version_info >= (3, 7):
+        @overload
+        def __init__(
+            self, fileobj: _ClosableZipStream, mode: str, zipinfo: ZipInfo, pwd: Optional[bytes], close_fileobj: Literal[True]
+        ) -> None: ...
+        @overload
         def __init__(
-            self, fileobj: IO[bytes], mode: str, zipinfo: ZipInfo, pwd: Optional[bytes] = ..., close_fileobj: bool = ...
+            self,
+            fileobj: _ClosableZipStream,
+            mode: str,
+            zipinfo: ZipInfo,
+            pwd: Optional[bytes] = ...,
+            *,
+            close_fileobj: Literal[True],
+        ) -> None: ...
+        @overload
+        def __init__(
+            self,
+            fileobj: _ZipStream,
+            mode: str,
+            zipinfo: ZipInfo,
+            pwd: Optional[bytes] = ...,
+            close_fileobj: bool = ...,
```

Comment:
I don't think this works as intended. If `fileobj` is a `_ZipStream` but not a `_ClosableZipStream`, and you pass `close_fileobj=True`, it will match this overload instead of erroring.

Reply:
See above: I think this is a reasonable tradeoff, considering our "prefer false negatives over false positives" policy.

Reply:
Then we might as well get rid of `_ClosableZipStream` altogether, because distinguishing between closable and non-closable zip streams doesn't affect whether the code type-checks.

Reply:
But having it separate has the advantage that the following fails:

```python
ZipExtFile(my_non_closable_stream, close_fileobj=True)
```

The only thing that can't be checked properly is the following:

```python
close: bool = True
ZipExtFile(my_non_closable_stream, close_fileobj=close)
```

Reply:
I don't think that fails. It doesn't match the first two overloads, but it does match the third overload which just requires any stream, and `bool` for `close_fileobj`.

Reply:
Good point. At this point I think that maybe the first solution was the best, and until python/mypy#6113 is fixed, people just use `# type: ignore`?

Reply:
Since we prefer false negatives, it might be better to just never require a `.close()` method.